xbmc/pvr/PVRDatabase.cpp

```diff
@@ -636,6 +636,18 @@ bool CPVRDatabase::Persist(CPVRChannelGroup &group)
   {
     CSingleLock lock(group.m_critSection);
 
+    /* Prevent AutoIncValue usage for internal groups as long they have fixed value */
+    if (group.GroupID() <= 0 && group.IsInternalGroup())
+      group.m_iGroupId = group.IsRadio() ?PVR_INTERNAL_GROUP_ID_RADIO:PVR_INTERNAL_GROUP_ID_TV;
+
+    /* First insert the # internal groups */
+    if(m_pDS->lastinsertid()==0)
+    {
+      strQuery = PrepareSQL("INSERT INTO channelgroups (idGroup) VALUES (NULL)");
+      ExecuteQuery(strQuery); //PVR_INTERNAL_GROUP_ID_RADIO
+      ExecuteQuery(strQuery); //PVR_INTERNAL_GROUP_ID_TV
```

Comment:
to ensure that the internal groups are stored at the very first time, we should create separate method which do all the necessary checks and insert the groups. BTW `m_pDS->lastinsertid()` returns 0 even if you didn't called a query and only returns the id from the last executed query.